src_common/common_utils.c

```diff
@@ -214,50 +214,25 @@ bool amountToString(const uint8_t *amount,
     return true;
 }
 
-bool getEthAddressFromKey(cx_ecfp_public_key_t *publicKey, uint8_t *out, cx_sha3_t *sha3Context) {
+void getEthAddressFromRawKey(const uint8_t raw_pubkey[static 65],
+                             uint8_t *out,
+                             cx_sha3_t *sha3Context) {
     uint8_t hashAddress[INT256_LENGTH];
-
-    if (cx_keccak_init_no_throw(sha3Context, 256) != CX_OK) {
-        return false;
-    }
-
-    if (cx_hash_no_throw((cx_hash_t *) sha3Context,
-                         CX_LAST,
-                         publicKey->W + 1,
-                         64,
-                         hashAddress,
-                         32) != CX_OK) {
-        return false;
-    }
-
+    CX_ASSERT(cx_keccak_init_no_throw(sha3Context, 256));
+    CX_ASSERT(
+        cx_hash_no_throw((cx_hash_t *) sha3Context, CX_LAST, raw_pubkey + 1, 64, hashAddress, 32));
```

Comment:
Should use `cx_keccak_256_hash` to simplify and spare RAM if at some point we changed the prototype (cx_sha3_t will be unused).

Reply:
Same below

Reply:
Done